### PR TITLE
iOS landscape rotation

### PR DESCRIPTION
**Pull request by JoelPickin**
### Description of Change ###

•	Register when the device is rotated by the OnElementPropertyChanged property of Height and Width.
•	When rotated set the bounds of the screen.
•	When rotated set the camera orientation of the capture connection.

### Bugs Fixed ###

This previously was a bug I raised myself but was understandably changed to a feature request.
#894

### API Changes ###

Added:
•	Added method in FormsCameraView.ios.cs: GetViewOrientation - Returns the AVCaptureVideoOrientation of the device.
•	Added method in FormsCameraView.ios.cs: SetCameraOrientation - Sets the previewLayer.Connection.VideoOrientation to the AVCaptureVideoOrientation from GetViewOrientation and then assigned the previewLayer.Connection to captureConnection.

Changed:
•	In method InitializeCamerain FormsCameraView.ios.cs added the line previewLayer.Connection.VideoOrientation = GetViewOrientation(); to set the correct orientation when the camera is being initialized.
•	In CameraViewRenderer.ios.cs - Added cases to switch in method OnElementPropertyChanged to check when Height & Width are changed ti trigger Control.SetBounds(Element.Width, Element.Height); & Control.SetCameraOrientation();

### Behavioral Changes ###

Rotating the device from landscape to the opposite landscape flips the camera view as the renderer still doesn't register the difference in rotation change. Rotating from portrait to landscape, then to portrait and landscape again works fine.

Images are captured in landscape also.

### PR Checklist ###

- [ ] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description) - Works in CameraView sample app
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs [hunk 1]
```diff
@@ -89,6 +89,32 @@ void SetStartOrientation()
 			}
 		}
 
+		public void SetCameraOrientation()
+		{
+			previewLayer.Connection.VideoOrientation = GetViewOrientation();
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs:94]:
```suggestion
			if (previewLayer.Connection != null)
				previewLayer.Connection.VideoOrientation = GetViewOrientation();
				
```

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs [hunk 2]
```diff
@@ -517,6 +545,8 @@ void InitializeCamera()
 
 				captureSession.CommitConfiguration();
 
+				previewLayer.Connection.VideoOrientation = GetViewOrientation();
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/FormsCameraView.ios.cs:548]:
```suggestion
				if (previewLayer.Connection != null)
					previewLayer.Connection.VideoOrientation = GetViewOrientation();
```

**Comment by jfversluis** [comment]
Great work @JoelPickin, thanks! I'm confused about this part:

> Rotating the device from landscape to the opposite landscape flips the camera view as the renderer still doesn't register the difference in rotation change. Rotating from portrait to landscape, then to portrait and landscape again works fine.

What does it mean? Reading the description it seems to work as you would expect? But the way you write the last part implies something different :D

**Comment by jfversluis** [comment]
@JoelPickin awaiting your answer, thanks!

**Comment by devopsoftware** [comment]
@JoelPickin  when this PR can be merged?

**Comment by JoelPickin** [comment]
> Great work @JoelPickin, thanks! I'm confused about this part:
> 
> > Rotating the device from landscape to the opposite landscape flips the camera view as the renderer still doesn't register the difference in rotation change. Rotating from portrait to landscape, then to portrait and landscape again works fine.
> 
> What does it mean? Reading the description it seems to work as you would expect? But the way you write the last part implies something different :D

Apologies for the late response. Life and new a job got in the way :)

I'll try my best to explain it, if not I will get some screenshots if I get time. 

It works fine if you rotate from portrait to landscape. The camera flips and it resizes. If I remember correctly the issue was if you go from landscape, flip it 180 degrees to the opposite landscape, then the screen rotates, but the camera view doesn't. So you end up seeing a camera view upside down. It only registers in the code when there is a resizing of the screen to reset the camera and its orientation. If you rotate the device back to portrait, let the screen rotate, then rotate to landscape, the camera will show correctly in the correct orientation.

A future way of solving this may be to register the change in the orientation through the gyroscope, so it detects any movement and can reset the camera that way.

Hope that answers your question. I will be available much more now to answer any question as quick as possible.

**Comment by GUOLDEV** [comment]
@jfversluis yes, #1524 should fix the rotation issue.

**Comment by GUOLDEV** [comment]
@jfversluis @JoelPickin the CameraView Sample is fixed as well, so it should be easy to confirm if it works 👍

**Comment by brminnick** [comment]
> versluis @JoelPickin the CameraView Sample is fixed as well, so it should be easy to confirm if it works 👍

@JoelPickin Could you confirm?

**Comment by WaldemarCichon** [comment]
Seems some tests are failing and also a rebase a needed

**Comment by WaldemarCichon** [comment]
I had compiled this changes but I get the following errors. How to avoid this (VS for Mac, newest version)

/Users/waldi/Projects/XamarinCommunityToolkit/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/CameraView/iOS/CameraViewRenderer.ios.cs(5,5): Error CS0152: The switch statement contains multiple cases with the label value '"Height"' (CS0152) (Xamarin.CommunityToolkit)

The same with Width

Regards

Waldemar

**Comment by bijington** [comment]
@WaldemarCichon we believe this issue should now be fixed in the latest release (1.3.0) but haven't been able to get confirmation from the author that it has been. 

Have you seen the same issue? If so are you able to confirm whether it has been fixed for us?

**Comment by WaldemarCichon** [comment]
> @WaldemarCichon we believe this issue should now be fixed in the latest release (1.3.0) but haven't been able to get confirmation from the author that it has been.
> 
> Have you seen the same issue? If so are you able to confirm whether it has been fixed for us?

Hi bjington,

yes i was facing the same issue. Currently downloaded the 1.3.0 and it is gone. Hooray. Thanks to the implementer, thanks to the reports.

Regards

Waldemar

**Comment by bijington** [comment]
@WaldemarCichon thabk you very much for confirming that the issue has been resolved. I shall close this issue down now.

@GUOLDEV big thanks for sorting this issue

**Comment by bijington** [comment]
Resolved in #1534

**Pull request closed without merging**